### ISSUE TITLE
Error when separate apt tasks   

### ISSUE THREAD

**Issue by varazir** [ansible/ansible#22124]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
apt

##### ANSIBLE VERSION

```
ansible 2.2.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### CONFIGURATION

No changes 

##### OS / ENVIRONMENT

Ubuntu 16.04.2 LTS (GNU/Linux 4.4.0-59-generic x86_64)

Both hosts and servers 

##### SUMMARY

Running 3 separate  atp actions failes on the third 

##### STEPS TO REPRODUCE

```yaml
```

https://gist.github.com/varazir/ba960bd117b9ee0e37e7810d3bacb10f

##### EXPECTED RESULTS

To remove unused dependency packages, 
Merging the 3 tasks it works fine. 

##### ACTUAL RESULTS

See Gist 

```

```

**Comment by bjolivot** [comment]
This playbook fail the same way

```yaml
- hosts: localhost
  become: true
  tasks:
    - name: Remove unused dependency packages
      apt:
        autoremove: yes
```

**Comment by bjolivot** [comment]
affect_2.3

**Comment by snowsky** [comment]
The packages variable are empty. I'll take a look to see where the problem is.

**Comment by gaddman** [comment]
Commit also fixes #21472
